core/modules/ccontrol/MergingHandler.h

```diff
@@ -45,6 +46,39 @@ namespace lsst {
 namespace qserv {
 namespace ccontrol {
 
+/// A class to delay creating the buffer until it is requested
+/// by xrootd SSI. The size of the buffer that will be needed is
+/// set using setTargetSize(int sz), and the buffer of that size is
+/// created by calling resizeToTargetSize(). When a buffer is no longer
+/// needed, zero() should be called to free the memory.
+class MergeBuffer {
+public:
+    using bufType = std::vector<char>;
+    MergeBuffer() {
+        _id = "MBI=" + std::to_string(_sequence++);
+        zero();
+    }
+    virtual ~MergeBuffer();
+
+    std::shared_ptr<bufType> getBuffer();
+    size_t getSize();
+    size_t getTargetSize() { return _targetSize; }
+    void setTargetSize(int sz);
+    void resizeToTargetSize();
```

Comment:
There's a lot of code that access the vector<char> underlying buffer directly, and any resize could invalidate the buffer. I don't want to risk an accidental resize, so simple getBuffer() never calls resize.